File: PHP/Interface/insert_new_music_track.php
<?
/* this script will handle the variables passed from the insert_customer.html file */

/* declare some relevant variables */

PRINT "<html>";
PRINT "<head><title>Insert music track</title></head>";
PRINT "<body>";

include 'connection.php';

$title = $_POST["title"];
$length = $_POST["length"];
$genre = $_POST["genre"];
$year = $_POST["year"];
$artist = $_POST["artist"];
$album = $_POST["album"];
$newartist = $_POST["newartist"];
$newalbum = $_POST["newalbum"];

/* MySQL table created to store the data */
$table0 = "music";
$table1 = "attribute";
$table2 = "genre";
$table3 = "year";
$table4 = "album";
$table5 = "artist";

$today = date("Y-m-d");
PRINT "<center>";
PRINT "Today is: $today.<br>";
PRINT "<hr style=\"margin-left:5em; margin-right:5em\">";
PRINT "You issued the following INSERT query:<br>";
PRINT "<ul>";
PRINT "<li>title: <em>$title</em>";
if($newartist != "new artist")
  {PRINT "<li>artist: <em>$newartist</em>";}
else
  {PRINT "<li>artist: <em>$artist</em>";}
if($newalbum != "new album")
  {PRINT "<li>album: <em>$newalbum</em>";}
else
  {PRINT "<li>album: <em>$album</em>";}
PRINT "<li>year: <em>$year</em>";
PRINT "<li>genre: <em>$genre</em>";
PRINT "<li>length: <em>$length</em>";
PRINT "</ul>";
PRINT "<hr style=\"margin-left:5em; margin-right:5em\">";
PRINT "Database server response:<br>"; 


/*insert year*/
$query = "SELECT year FROM $table3 WHERE year = '$year'";
$result = MYSQL_QUERY($query);

/* How many rows in the result? */
if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}

if($number == 0)
{
  /* Insert year into table year */
  $query = "INSERT INTO $table3 VALUES('null', '$year')";
  $result = MYSQL_QUERY($query);

  if($result != 0)
  {
    $affected_rows =  mysql_affected_rows();
    PRINT "$table3: ";
    PRINT "<strong>Query OK. Rows affected $affected_rows</strong><br>";
  }
  else
  {
     PRINT "$table3: ";
     PRINT "<strong>Query FAILED. Probably you don't have the proper access rights!</strong><br>";
  }
}

/*insert genre*/
$query = "SELECT genre FROM $table2 WHERE genre = '$genre'";
$result = MYSQL_QUERY($query);

/* How many rows in the result? */
if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}

if($number == 0)
{
  /* Insert genre into table genre */
  $query = "INSERT INTO $table2 VALUES('null', '$genre')";
  $result0 = MYSQL_QUERY($query);

  if($result0 != 0)
  {
    $affected_rows =  mysql_affected_rows();
    PRINT "$table2: ";
    PRINT "<strong>Query OK. Rows affected $affected_rows</strong><br>";
  }
  else
  {
     PRINT "$table2: ";
     PRINT "<strong>Query FAILED. Probably you don't have the proper access rights!</strong><br>";
  }
}

/*insert artist*/
if($newartist != "new artist")
{
  $query = "SELECT name FROM $table5 WHERE name = '$newartist'";
  $result = MYSQL_QUERY($query);

  /* How many rows in the result? */
  if(!$result){
    $number = 0;
  }else{
    $number = MYSQL_NUMROWS($result);
  }

  if($number == 0)
  {
    /* Insert name into table artist */
    $query = "INSERT INTO $table5 VALUES('null', '$newartist')";
    $result = MYSQL_QUERY($query);

    if($result != 0)
    {
      $affected_rows =  mysql_affected_rows();
      PRINT "$table5: ";
      PRINT "<strong>Query OK. Rows affected $affected_rows</strong><br>";
    }
    else
    {
       PRINT "$table5: ";
       PRINT "<strong>Query FAILED. Probably you don't have the proper access rights!</strong><br>";
    }
  }
}

/*insert album*/
if($newalbum != "new album")
{
  $query = "SELECT name FROM $table4 WHERE name = '$newalbum'";
  $result = MYSQL_QUERY($query);

  /* How many rows in the result? */
  if(!$result){
    $number = 0;
  }else{
    $number = MYSQL_NUMROWS($result);
  }

  if($number == 0)
  {
    /* Insert name into table artist */
    $query = "INSERT INTO $table4 VALUES('null', '$newalbum')";
    $result = MYSQL_QUERY($query);

    if($result != 0)
    {
      $affected_rows =  mysql_affected_rows();
      PRINT "$table4: ";
      PRINT "<strong>Query OK. Rows affected $affected_rows</strong><br>";
    }
    else
    {
       PRINT "$table4: ";
       PRINT "<strong>Query FAILED. Probably you don't have the proper access rights!</strong><br>";
    }
  }
}

/* Get yearid from table year */
$query = "SELECT yid FROM $table3 WHERE year = '$year'";
$result = MYSQL_QUERY($query);

if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}

if ($number == 0)
{
  PRINT "$table3: ";
  PRINT "<strong>Query FAILED. Probably you don't have the proper access rights!</strong><br>";
}
else
{
  $yid = mysql_result($result, "0", "yid");
}

/* Get gid from table genre */
$query = "SELECT gid FROM $table2 WHERE genre = '$genre'";
$result = MYSQL_QUERY($query);

if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}

if ($number == 0)
{
  PRINT "$table2: ";
  PRINT "<strong>Query FAILED. Probably you don't have the proper access rights!</strong><br>";
}
else
{
  $gid = mysql_result($result, "0", "gid");
}

/* Get artid from table artist */
if($newartist != "new artist")
  {$query = "SELECT artid FROM $table5 WHERE name = '$newartist'";}
else
  {$query = "SELECT artid FROM $table5 WHERE name = '$artist'";}
$result = MYSQL_QUERY($query);

if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}

if ($number == 0)
{
  PRINT "$table5: ";
  PRINT "<strong>Query FAILED. Probably you don't have the proper access rights!</strong><br>";
}
else
{
  $artid = mysql_result($result, "0", "artid");
}

/* Get albid from table album */
if($newalbum != "new album")
  {$query = "SELECT albid FROM $table4 WHERE name = '$newalbum'";}
else
  {$query = "SELECT albid FROM $table4 WHERE name = '$album'";}
$result = MYSQL_QUERY($query);

if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}

if ($number == 0)
{
  PRINT "$table4: ";
  PRINT "<strong>Query FAILED. Probably you don't have the proper access rights!</strong><br>";
}
else
{
  $albid = mysql_result($result, "0", "albid");
}

/* Insert aid, gid, yid, artid, albid, length into table attributes */
$query = "INSERT INTO $table1 VALUES('null', '$gid', '$yid', '$artid', '$albid', '$length')";
$result = MYSQL_QUERY($query);

if($result != 0)
{
  $affected_rows =  mysql_affected_rows();
  PRINT "$table1: ";
  PRINT "<strong>Query OK. Rows affected $affected_rows</strong><br>";
}
else
{
   PRINT "$table1: ";
   PRINT "<strong>Query FAILED. Probably you don't have the proper access rights!</strong><br>";
}

/* Get aid from table attribute */
$query = "SELECT aid FROM $table1 WHERE gid = '$gid' and yid = '$yid' and artid = '$artid' and albid = '$albid' and length = '$length'";
$result = MYSQL_QUERY($query);

if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}

if ($number == 0)
{
  PRINT "$table1: ";
  PRINT "<strong>Query FAILED. Probably you don't have the proper access rights!</strong><br>";
}
else
{
  $aid = mysql_result($result, "0", "aid");
}

/* Insert mid, aid, title into table music */
$query = "INSERT INTO $table0 VALUES('null', '$aid', '$title')";
$result = MYSQL_QUERY($query);

if($result != 0)
{
  $affected_rows =  mysql_affected_rows();
  PRINT "$table0: ";
  PRINT "<strong>Query OK. Rows affected $affected_rows</strong><br>";
}
else
{
   PRINT "$table0: ";
   PRINT "<strong>Query FAILED. Probably you don't have the proper access rights!</strong><br>";
}

PRINT "</center>";
PRINT "<hr style=\"margin-left:5em; margin-right:5em\">";
PRINT "<center>";
PRINT "<a href = \"insert_new_music_track_interface.php\">Back</a>";
PRINT "</center>";
PRINT "</body>";
PRINT "</html>";

/* Close the database connection */
MYSQL_CLOSE();

?>

File: Servlets/Interface/db2/src/java/dbase2/musicdb/insert_new_music_track.java
package dbase2.musicdb;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletResponse;

import java.io.IOException;
import java.io.PrintWriter;

import java.sql.Connection;
import java.sql.SQLException;
import java.sql.Statement;
import java.sql.DriverManager;
import java.sql.ResultSet;

public class insert_new_music_track extends HttpServlet{
  
  public void doGet(HttpServletRequest request, HttpServletResponse response)
    throws ServletException, IOException{
    response.setContentType("text/html");
    PrintWriter writer = response.getWriter();
    writer.write("<html>\n");
    writer.write("<head><title>Insert Music Track</title></head>\n");
    writer.write("<body>\n");
    writer.write("<center>\n");
    
    String title = request.getParameter("title");
    if((title == null) || (title.length() == 0)){
      printMsg("Can not insert a title with no value!", writer, request);
      return;
    }
    
    String length = request.getParameter("length");
    if((length == null) || (length.length() == 0)){
      printMsg("Can not insert a length with no value!", writer, request);
      return;
    }
    
    String genre = request.getParameter("genre");
    if((genre == null) || (genre.length() == 0)){
      printMsg("Can not insert a genre with no value!", writer, request);
      return;
    }
    
    String year = request.getParameter("year");
    if((year == null) || (year.length() == 0)){
      printMsg("Can not insert a year with no value!", writer, request);
      return;
    }
    
    String artist = request.getParameter("artist");
    if((artist == null) || (artist.length() == 0)){
      printMsg("Can not insert an artist with no value!", writer, request);
      return;
    }
    
    String album = request.getParameter("album");
    if((album == null) || (album.length() == 0)){
      printMsg("Can not insert an album with no value!", writer, request);
      return;
    }
    
    String newartist = request.getParameter("newartist");
    if((newartist == null) || (newartist.length() == 0)){
      printMsg("Can not insert an newartist with no value!", writer, request);
      return;
    }
    
    String newalbum = request.getParameter("newalbum");
    if((newalbum == null) || (newalbum.length() == 0)){
      printMsg("Can not insert an newalbum with no value!", writer, request);
      return;
    }
    
    writer.write("<hr style=\"margin-left:5em; margin-right:5em\">\n");
    writer.write("You issued the following INSERT query:<br>\n");
    writer.write("<ul>\n");
    writer.write("<li>title: <em>"+title+"</em>\n");
    
    if(newartist != "new artist")
      writer.write("<li>artist: <em>"+newartist+"</em>\n");
    else
      writer.write("<li>artist: <em>"+artist+"</em>\n");
    if(newalbum != "new album")
      writer.write("<li>album: <em>"+newalbum+"</em>\n");
    else
      writer.write("<li>album: <em>"+album+"</em>\n");
    writer.write("<li>year: <em>"+year+"</em>\n");
    writer.write("<li>genre: <em>"+genre+"</em>\n");
    writer.write("<li>length: <em>"+length+"</em>\n");
    writer.write("</ul>\n");
    writer.write("<hr style=\"margin-left:5em; margin-right:5em\">\n");
    writer.write("Database server response:<br>\n");
    
    try{
      Class.forName("com.mysql.jdbc.Driver");
    }catch(ClassNotFoundException exc){
      exc.printStackTrace();
      printMsg("No JDBC driver found!", writer, request);
      return;
    }
    try{
      Connection connection = DriverManager.getConnection("jdbc:mysql://localhost/db2", "student", "student");
      Statement statement = connection.createStatement(); //referential integrity
      ResultSet result = null;
      
       /*insert year*/
       result = statement.executeQuery("SELECT year FROM year WHERE year = "+Integer.parseInt(year));
      
       if (result == null)
       {
          /* Insert year into table year */
          String insert_sql_stmt = "INSERT INTO year VALUES('null', '"+Integer.parseInt(year)+"')";
          statement = connection.createStatement();
          int row = statement.executeUpdate(insert_sql_stmt);
          writer.write("<strong>Year inserted!</<strong><br>\n");
        }
        
       /*insert genre*/
       result = statement.executeQuery("SELECT genre FROM genre WHERE genre = "+genre+"");
      
       if (result == null)
       {
          /* Insert genre into table genre */
          String insert_sql_stmt = "INSERT INTO genre VALUES('null', '"+genre+"')";
          statement = connection.createStatement();
          int row = statement.executeUpdate(insert_sql_stmt);
          writer.write("<strong>Genre inserted!</<strong><br>\n");
       }
       
    if(newartist != "new artist")
    {
       /*insert artist*/
       result = statement.executeQuery("SELECT name FROM artist WHERE name = "+newartist+"");
      
       if (result == null)
       {
          /* Insert genre into table artist */
          String insert_sql_stmt = "INSERT INTO artist VALUES('null', '"+artist+"')";
          statement = connection.createStatement();
          int row = statement.executeUpdate(insert_sql_stmt);
          writer.write("<strong>Artist inserted!</<strong><br>\n");
       }
     }

    if(newalbum != "new album")
    {
       /*insert album*/
       result = statement.executeQuery("SELECT name FROM album WHERE name = "+newalbum+"");
      
       if (result == null)
       {
          /* Insert genre into table genre */
          String insert_sql_stmt = "INSERT INTO album VALUES('null', '"+album+"')";
          statement = connection.createStatement();
          int row = statement.executeUpdate(insert_sql_stmt);
          writer.write("<strong>Album inserted!</<strong><br>\n");
       }
     }

    /* Get yearid from table year */
    result = statement.executeQuery("SELECT yid FROM year WHERE year = '"+Integer.parseInt(year)+"'");

    int yid = 0, gid = 0, artid = 0, albid = 0, aid = 0;
    
     if (result != null)
       yid = new Integer(result.getInt("yid"));
     
    /* Get gid from table genre */
    result = statement.executeQuery("SELECT gid FROM genre WHERE year = '"+genre+"'");

     if (result != null)
       gid = new Integer(result.getInt("gid"));

    /* Get artid from table artist */
    if(newartist != "new artist")
      result = statement.executeQuery("SELECT artid FROM artist WHERE name = '"+newartist+"'");
    else
      result = statement.executeQuery("SELECT artid FROM artist WHERE name = '"+artist+"'");

     if (result != null)
       artid = new Integer(result.getInt("artid"));

    /* Get albid from table album */
    if(newalbum != "new album")
      result = statement.executeQuery("SELECT albid FROM album WHERE name = '"+newalbum+"'");
    else
      result = statement.executeQuery("SELECT albid FROM album WHERE name = '"+album+"'");

     if (result != null)
       albid = new Integer(result.getInt("album"));

    /* Insert aid, gid, yid, artid, albid, length into table attributes */
    String insert_sql_stmt = "INSERT INTO attribute VALUES('null', '"+gid+"', '"+yid+"', '"+artid+"', '"+albid+"', '"+Integer.parseInt(length)+"')";
    statement = connection.createStatement();
    int row = statement.executeUpdate(insert_sql_stmt);
    writer.write("<strong>Attribute inserted!</<strong><br>\n");


    /* Get aid from table attribute */
    result = statement.executeQuery("SELECT aid FROM attribute WHERE gid = '"+gid+"' and yid = '"+yid+"' and artid = '"+artid+"' and albid = '"+albid+"' and length = '"+length+"'");

     if (result != null)
       aid = new Integer(result.getInt("aid"));


    /* Insert mid, aid, title into table music */
    insert_sql_stmt = "INSERT INTO music VALUES('null', '"+aid+"', '"+title+"')";
    statement = connection.createStatement();
    row = statement.executeUpdate(insert_sql_stmt);
    writer.write("<strong>music inserted!</<strong><br>\n");
    writer.write("</center>\n");
    printMsg("", writer, request);
      }catch(SQLException exc){
        exc.printStackTrace();
        printMsg("database error!", writer, request);
      }
}

  private void printMsg(String msg, PrintWriter writer, HttpServletRequest request){
    writer.write("<h1>" + msg + "</h1>\n");
    writer.write("</center>\n");
    writer.write("<hr style=\"margin-left:5em; margin-right:5em\">\n");
    writer.write("<center>\n");
    writer.write("<a href = \"\\dbase2\\insert_new_music_track_interface\">Back</a>\n");
    writer.write("</center>\n");
    writer.write("</body>\n");
    writer.write("</html>\n");
  }
}

